Zend/zend_portability.h

```diff
@@ -44,6 +44,13 @@
 # define ZEND_PATHS_SEPARATOR		':'
 #endif
 
+/* This is a work around for GCC bug 69602: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69602 */
+#if defined(EWOULDBLOCK) && EAGAIN != EWOULDBLOCK
+# define IS_TRANSIENT_ERROR(err) (err == EAGAIN || err == EWOULDBLOCK)
+#else
+# define IS_TRANSIENT_ERROR(err) (err == EAGAIN)
+#endif
+
```

Comment:
Maybe you could put the code here, it also works on ext/sockets

https://github.com/php/php-src/blob/62c6d6952ead055b2a08636eacdce6bb74168de1/main/php_network.h#L35-L50

BTW, the name is inspired by [libuv](https://github.com/libuv/libuv/blob/v1.x/src/unix/stream.c#L73), it defined on the C source file instead of the header file, so it's better to add a prefix on it, or named like ZEND_IS_EAGAIN (I am not sure, naming is not my strength...)